chore(design-tokens): add sizing design tokens

rinaok opened the pull request.

YonatanKra commented on globals/sizing/consts.js:
```diff
@@ -0,0 +1,3 @@
+export const SAVE_PATH = 'dist/sizing/sizing.json';
+export const subItems = ['4px-scale', '8px-scale'].map(val => `${val}`);
+export const properties = ['xxxs', 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl', 'xxxl'].map(val => `${val}`);
```
There's no need for mapping if you return the same value.

YonatanKra commented on globals/sizing/consts.js:
```diff
@@ -0,0 +1,3 @@
+export const SAVE_PATH = 'dist/sizing/sizing.json';
+export const subItems = ['4px-scale', '8px-scale'].map(val => `${val}`);
```
Same comment as below for no need for mapping.

YonatanKra reviewed (changes requested):
Additionally to the two comments, could you please add tests like the shadow tests?
There are tests both for the parser and the global scripts.

rachelbt commented on scripts/figma-parsers/parsers/sizing-figma-data-parser.js:
```diff
@@ -0,0 +1,76 @@
+    SCALES_NAMES = ['4px scale', '8px scale'],
+    CHILD_TYPE = 'FRAME',
+    LEVELS = {
+        'xxxs': 'xxxs',
```
I think we should consider using xs, 2xs, 3xs, xl, 2xl, 3xl ext.to avoid spelling and make it more readable

rachelbt reviewed (commented):
I think we should consider using xs, 2xs, 3xs, xl, 2xl, 3xl ext.to avoid spelling and make it more readable

YonatanKra commented:
> I think we should consider using xs, 2xs, 3xs, xl, 2xl, 3xl ext.to avoid spelling and make it more readable

Sounds reasonable. @rachelbt do you have any examples of libraries that use either convention?

rachelbt commented:
> > I think we should consider using xs, 2xs, 3xs, xl, 2xl, 3xl ext.to avoid spelling and make it more readable
> 
> Sounds reasonable. @rachelbt do you have any examples of libraries that use either convention?

Its usually in use for fonts, but not only:

- https://uniformcss.com/cheatsheet/font-sizes/
- https://windicss.org/utilities/sizing.html
- https://tailwindcss.com/docs/max-width

YonatanKra commented:
> > > I think we should consider using xs, 2xs, 3xs, xl, 2xl, 3xl ext.to avoid spelling and make it more readable
> > 
> > Sounds reasonable. @rachelbt do you have any examples of libraries that use either convention?
> 
> Its usually in use for fonts, but not only:
> 
> - https://uniformcss.com/cheatsheet/font-sizes/
> - https://windicss.org/utilities/sizing.html
> - https://tailwindcss.com/docs/max-width
> 

Oh no!!! Now that you said `tailwind` is doing it @yinonov will never agree 😉

yinonov commented:
predefining by size units will result in the following interface

```html
<vwc-icon size="3xs" ...
```

unless we map each component to its 3 preferred values 
e.g. icon size could map -
- small = 3xs
- medium = xl
- large = 2xl

alternatively we can refine the naming above to a readable version have a "wild" interface
e.g. icon size picks types and reacts only to that pick -
- 3xs
- xl
- 2xl

rinaok commented:
> Additionally to the two comments, could you please add tests like the shadow tests? There are tests both for the parser and the global scripts.

Added tests and changed the sizes to xs, 2xs, 3xs, xl, 2xl, 3xl

jshenkman commented:
> predefining by size units will result in the following interface
> 
> ```
> <vwc-icon size="3xs" ...
> ```
> 
> unless we map each component to its 3 preferred values e.g. icon size could map -
> 
> * small (3xs)
> * medium (xl)
> * large (2xl)
> 
> alternatively we can refine the naming above to a readable version have a "wild" interface e.g. icon size picks types and reacts only to that pick -
> 
> * 3xs
> * xl
> * 2xl

I think mapping each component to its 3 sizes actually makes sense. "Hiding" the values themself and just giving the scale.
Kinda similar to color token we hide the hardcoded values under connotation, so the user knows what information he wants to show but doesn't care what shade of purple is the CTA one. Same with button size, the user needs the smallest one so he will choose small, he doesn't need to care/know that it's 32px hight

jshenkman commented:
> > Additionally to the two comments, could you please add tests like the shadow tests? There are tests both for the parser and the global scripts.
> 
> Added tests and changed the sizes to xs, 2xs, 3xs, xl, 2xl, 3xl

I'm happy with this naming! Updated also in Figma file

yinonov commented:
> tton size, the user needs the smallest one so he will choose small, he doesn't need to care/know that it's 32px hight

so last question - hypothetically textfield 'small' can be different than button 'small'. is that an ok way to go (ui-wise)?

jshenkman commented:
> > tton size, the user needs the smallest one so he will choose small, he doesn't need to care/know that it's 32px hight
> 
> so last question - hypothetically textfield 'small' can be different than button 'small'. is that an ok way to go (ui-wise)?

Umm good Q, but it's pretty much the same today using sizing names - dense, default, and enlarged. 
Today we have 48px text field which is default and then 40px which is dense as opposed to the button which is dense 32 default 40 and enlarged 48... so even now it's confusing for the user, he will want a default button and default text field and get different sizes...

YonatanKra commented on test/sizing/sizing.spec.js:
```diff
@@ -0,0 +1,39 @@
+        });
+
+        describe(`Sizing`, function () {
+            it(`should write to JSON`, function () {
```
`should write to JSON only once`

YonatanKra commented on test/sizing/sizing.spec.js:
```diff
@@ -0,0 +1,40 @@
+        });
+
+        describe(`Sizing`, function () {
+            it(`should write to JSON once`, function () {
```
`should write to JSON only once` and make sure the mockWriteJSON is called only once.
`mockWriteJson = jest.fn(mockWriteJson);`
And then when you want to check, do this:
`expect(mockJsonWrite.mock.calls).toHaveLength(1);`
Look at the `color.spec` to see an example.

YonatanKra commented on test/sizing/sizing.spec.js:
```diff
@@ -0,0 +1,46 @@
+
+        describe(`Sizing`, function () {
+            it(`should write to JSON only once`, function () {
+                mockWriteJson = jest.fn(mockWriteJson);
```
Do this at the top. Something like:
```
beforeEach(function() {
  results.length = 0;
  mockWriteJson = jest.fn(function(data, path) {
    results.push({
      data, path
    });
});
```

YonatanKra commented on test/sizing/sizing.spec.js:
```diff
@@ -0,0 +1,45 @@
+        });
+
+        describe(`Sizing`, function () {
+            it(`should write to JSON only once`, function () {
+                sizingParser.parse(rawData, mockWriteJson);
+                expect(mockWriteJson.mock.calls).toHaveLength(1);
+            });
+
+            it(`should write to JSON with length equal to 1`, function () {
+                sizingParser.parse(rawData, mockWriteJson);
+                expect(results.length).toEqual(1);
+            });
```
Aren't these two do the same?
